.changeset/wet-eggs-rule.md

```diff
@@ -0,0 +1,30 @@
+---
+'@graphql-hive/cli': minor
+---
+
+Add `--target` flag for commands `app:create`, `app:publish`, `operations:check`, `schema:check`,
+`schema:delete`, `schema:fetch`, `schema:publish` and `dev`.
+
+The `--target` flag can be used to specify the target on which the operation should be performed.
+Either a slug or ID of the target can be provided.
+
+A provided slug must follow the format `$organizationSlug/$projectSlug/$targetSlug` (e.g.
+`the-guild/graphql-hive/staging`).
+
+**Example using target slug**
+
+```bash
+hive schema:publish --target the-guild/graphql-hive/production ./my-schema.graphql
+```
+
+A target id, must be a valid target UUID.
+
+**Example using target id**
+
+```bash
+hive schema:publish --target a0f4c605-6541-4350-8cfe-b31f21a4bf80 ./my-schema.graphql
```

Comment:
side-node: should we add an issue to improve how the dashboard shows IDs to the user? 🤔

Reply:
I think that's a good idea. Also having a set of commands displayed for publishing your first schema would be helpful:
<img width="901" alt="Screenshot 2025-02-03 at 8 40 50 AM" src="https://github.com/user-attachments/assets/ec7c8d05-e0e7-4ec3-8515-5fe6fde7f006" />

Reply:
also created https://github.com/graphql-hive/console/issues/6466